lib/logger.js

```diff
@@ -522,83 +614,104 @@ class Logger extends EventEmitter {
       ? buffer[buffer.length - 1].line
       : null
 
-    axios(config)
-      .then((response) => {
-        // We have a 200-level success code.
-        const totalLinesSent = buffer.length
-        this[kIsLoggingBackedOff] = false
-        this[kAttempts] = 0
-        this[kIsSending] = false
-        this[kTotalLinesReady] -= totalLinesSent
-        this[kReadyToSend].shift()
-
-        // Assist GC by killing the buffer and removing it from readyToSend
-        buffer.length = 0
-
-        this.emit('send', {
-          httpStatus: response.status
-        , firstLine
-        , lastLine
-        , totalLinesSent
-        , totalLinesReady: this[kTotalLinesReady]
-        , bufferCount: this[kReadyToSend].length
-        })
-
-        if (this[kReadyToSend].length) {
-          // Continue to send any backed up payloads that have accumulated
-          this.send()
-          return
-        }
-
-        this.emit('cleared', {
-          message: ALL_CLEAR_SENT
-        })
-      })
-      .catch((error) => {
-        const code = error.response
-          ? error.response.status
-          : error.code // timeouts will populate this
+    const data = stringify({e: 'ls', ls: buffer})
 
-        const retrying = ERROR_CODES_TO_RETRY.has(code)
-        const err = new Error('An error occured while sending logs to the cloud.')
+    this._getSendPayload(data, (error, payload) => {
+      if (error) {
+        const err = new Error('Error gzipping data')
         err.meta = {
-          actual: error.message
-        , code
-        , firstLine
-        , lastLine
-        , retrying
-        , attempts: ++this[kAttempts]
+          message: 'Will attempt to send data uncompressed'
+        , error
         }
+        // We will still send, but without compression
+        delete config.headers['Content-Encoding']
```

Comment:
I thought about this.  I guess I didn't consider the deopt important because this is a one-time-use object.  There won't be any lookups after this point where we'd suffer map transition lookups.  Should we do it the suggested way just for consistency?

Reply:
true, yeah maybe the better argument is consistency - `delete` just sets off those alarms as a thing to avoid as a strict rule.

Reply:
nm, you're right.  Done.